### PR TITLE
optimize grpc relative code

### PR DESCRIPTION
**Pull request by gaoxinge**
**What this PR does**:

- fix typo
- update grpc from `v1.26.0` to `v1.29.1`
- update etcd from `v3.3.25+incompatible` to `v0.0.0-20200402134248-51bdeb39e698`
- directly [SetProxyImpl](https://github.com/dubbo-x/dubbo-go/blob/3.0-grpc-test/protocol/grpc/server.go#L154) instead of using [reflect](https://github.com/dubbo-x/dubbo-go/blob/3.0/protocol/grpc/server.go#L159)
- fix service discovery test

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

**Comment by gaoxinge** [comment]
@AlexStocks @LaurenceLiZhixin @wenxuwan Please help review the code.

- There are some unit tests that can't pass because of service discovery. I am not able to fix them.
- I mainly modify the files that includes grpc, etcd and go.mod. It seems that I modify many files for many conflicts  to resolve because I and others try to fix the same unit test about metadata and service discovery.

**Review by LaurenceLiZhixin — APPROVED** [review]

File: metadata/service/inmemory/service_proxy.go [hunk 1]
```diff
@@ -64,10 +64,10 @@ func (m *MetadataServiceProxy) GetExportedURLs(serviceInterface string, group st
 		return []*common.URL{}, nil
 	}
 
-	urlStrs := res.Result().([]string)
+	urlStrs := *res.Result().(*[]interface{})
```
Comment (AlexStocks) [inline metadata/service/inmemory/service_proxy.go]: pls add comment for this change

**Review by zouyx — COMMENTED** [review]
make checking success pls

File: metadata/service/inmemory/service_proxy.go (hunk 1 above)
Comment (gaoxinge) [reply metadata/service/inmemory/service_proxy.go]: @AlexStocks @LaurenceLiZhixin This code is outdated. I have keep it same as 3.0 branch.

**Comment by AlexStocks** [comment]
have u finished ur pr?

**Comment by gaoxinge** [comment]
@AlexStocks @LaurenceLiZhixin @wenxuwan @zouyx Help review the code.

I mainly：
- modify the files that includes grpc, etcd and go.mod to update grpc and etcd
- fix the service discovery for consul, nacos and zookeeper

It seems that I modify many files for many conflicts to resolve because I and others try to fix the same unit test about metadata and service discovery.

**Comment by gaoxinge** [comment]
This is unit test fail with [cluster](https://github.com/apache/dubbo-go/pull/1160/checks?check_run_id=2447866081#step:9:1169).

**Pull request closed without merging**